recipes/icu/all/conanfile.py

```diff
@@ -119,6 +119,11 @@ def generate(self):
         if cross_building(self):
             base_path = unix_path(self, self.dependencies.build["icu"].package_folder)
             tc.configure_args.append(f"--with-cross-build={base_path}")
+            if (not is_msvc(self)):
+                # --with-cross-build above prevents tc.generate() from setting --build option.
+                # Workaround for https://github.com/conan-io/conan/issues/12642
+                gnu_triplet = get_gnu_triplet(str(self._settings_build.os), str(self._settings_build.arch))
```

Comment:
I think it's broken on windows (not msvc), get_gnu_triplet needs compiler.

Reply:
I just made it consistent with `elif is_msvc(self):` couple of lines below, so build is not set twice for msvc.

Reply:
I know, just pass host compiler as third argument if build machine is windows

Reply:
Ah, ok, understood